### DISCUSSION TITLE
Possible Performance Regression RaycastVehicle

### DISCUSSION THREAD

**Discussion by EthanShoeDev** [pmndrs/use-cannon#420] [Q&A] [answered]
I get much better performance using this example from codepen:
[https://codesandbox.io/s/raycast-vehicle-forked-srbmru](https://codesandbox.io/s/raycast-vehicle-forked-srbmru)
than the example in the repo:
[https://cannon.pmnd.rs/#/demo/RaycastVehicle](https://cannon.pmnd.rs/#/demo/RaycastVehicle)

Most notable is the difference in input lag. 
Does anyone know what might be causing the issue?

**Comment by EthanShoeDev** [comment, marked as answer]
For the next reader:

I found the main source of lag was in the vehicle model.
After removing shadows, the spotlight, and hard to render materials the responsiveness improved dramatically.

Here is an example branch that has those changes:
https://github.com/EthanShoeDev/use-cannon/tree/performance-raycast-vehicle


I think this is a rather important issue as I was evaluating if this library would be suitable for a game.
Based on the initial performance of the demo, I thought there was no way this package would work.
I initially thought the lag was being introduced via the web worker architecture and that cannot be disabled within this library.

I find it likely that others would come to the same conclusion simply based on the example performance.

I am not nearly as worried about it if all I need to do is make sure to use easy to render materials.

**Reply by isaac-mason** [reply]
Thanks for posting the follow up! We should probably look to make the demo more performant as you've described then

**Reply by EthanShoeDev** [reply]
Even a "performance" mode that you can toggle into would be awesome. Just something to demonstrate the library itself is not at fault. 💪
